__tests__/api/id/form/bearer.test.ts

```diff
@@ -32,131 +31,374 @@ describe("/id/[form]/bearer", () => {
-      (prismaMock.template.findUnique as jest.MockedFunction<any>).mockResolvedValue({
-        bearerToken: null,
+      it("Should return a 200 status code and a valid token if it exists for a form.", async () => {
+        // Mocking executeQuery to return a valid bearer token
+
+        (prismaMock.template.findUnique as jest.MockedFunction<any>).mockResolvedValue({
+          bearerToken: "testBearerToken",
+          users: [{ id: "1" }],
+        });
+        const { req, res } = createMocks({
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          query: {
+            form: "12",
+          },
+        });
+
+        await retrieve(req, res);
+        expect(res.statusCode).toBe(200);
+        expect(JSON.parse(res._getData())).toEqual(
+          expect.objectContaining({ bearerToken: "testBearerToken" })
+        );
       });
 
-      const { req, res } = createMocks({
-        method: "GET",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        query: {
-          form: "11",
-        },
+      it("Should return 404 status code if no form was found", async () => {
+        // Mocking executeQuery to return an empty array
+        (prismaMock.template.findUnique as jest.MockedFunction<any>).mockResolvedValue(null);
+
+        const { req, res } = createMocks({
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          query: {
+            form: "23",
+          },
+        });
+        await retrieve(req, res);
+
+        expect(res.statusCode).toBe(404);
+        expect(JSON.parse(res._getData()).error).toEqual("Not Found");
       });
 
-      await retrieve(req, res);
-      expect(JSON.parse(res._getData())).toEqual(expect.objectContaining({ bearerToken: null }));
-      expect(res.statusCode).toBe(200);
+      it("Should return a 404 status code if there's an unexpected Prisma error", async () => {
+        // Mocking executeQuery to throw an error
+
+        prismaMock.template.findUnique.mockRejectedValue(
+          new Prisma.PrismaClientKnownRequestError("Timed out", "P2024", "4.3.2")
+        );
+        const { req, res } = createMocks({
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          query: {
+            form: "101",
+          },
+        });
+
+        await retrieve(req, res);
+        expect(res.statusCode).toBe(404);
+        expect(JSON.parse(res._getData()).error).toEqual("Not Found");
+      });
     });
-
-    it("Should return a 200 status code and a valid token if it exists for a form.", async () => {
-      // Mocking executeQuery to return a valid bearer token
-
-      (prismaMock.template.findUnique as jest.MockedFunction<any>).mockResolvedValue({
-        bearerToken: "testBearerToken",
+    describe("ManageForms Permissions", () => {
```

Comment:
In one of your previous pull request you have used `test.each` in order to test multiple http methods for a same test. Based on that I have used `describe.each` ([while I was adding tests to users and templates](https://github.com/cds-snc/platform-forms-client/commit/d7ed658069cb2c044681440f5eebf045045a334c#diff-ddb612901d321b473788904cf6b078ff5bc7168860ac034733876dc21bf0a3acR66)) in order to run the same tests on different privileges. 
If you like it I could take care of doing a little refactoring to these tests before making my own last pull request for permissions.

Reply:
It would be great If you could refactor for those that are common and should have the same result for each.  Initially why I split them out was because some of same tests should fail between Base and ManageForms privileges and I wasn't sure how I was going to organize them in the test suite.